src/replica/SqlRowStatsJob.cc

```diff
@@ -0,0 +1,332 @@
+        }
+    }
+    finish(lock, extendedState);
+}
+
+
+bool SqlRowStatsJob::_process(
+        string const& context_, bool isPartitioned, SqlJobResult::Worker const& worker,
+        SqlJobResult::Scope const& internalTable, SqlResultSet::ResultSet const& resultSet,
+        map<string, map<TransactionId, vector<size_t>>>& counters) const {
+
+    bool const succeeded = true;
+    auto const reportResultThatHas = [&] (string const& problem) {
+        LOGS(_log, LOG_LVL_ERROR, context_ << "result set received from worker '"
+                << worker << "' for table '" << internalTable << "' has " << problem);
+    };
+
+    // This scenario is possible in the FORCED mode. And we're guaranteed
+    // not to seen it in the ENABLED mode.
+    if (resultSet.extendedStatus != ProtocolStatusExt::NONE) return succeeded;
+
+    // Skip special tables.
+    if (isPartitioned) {
+        if (internalTable == table()) return succeeded;
+        try {
+            ChunkedTable const chunkedTable(internalTable);
+            if (chunkedTable.chunk() == lsst::qserv::DUMMY_CHUNK) return succeeded;
+            if (chunkedTable.baseName() != table()) {
+                reportResultThatHas("incorrect base name of the partitioned table");
+                return !succeeded;
+            }
+        } catch (...) {
+            reportResultThatHas("incorrect name of the partitioned table");
+            return !succeeded;
+        }
+    }
+
+    // Expecting a result set that have least one row and exactly two columns:
+    //
+    //   'qserv_trans_id' | 'num_rows'
+    //  ------------------+------------
+    //
+    if (resultSet.fields.size() != 2 || resultSet.fields[0].name != "qserv_trans_id" ||
+        resultSet.fields[1].name != "num_rows") {
+        reportResultThatHas("unexpected format");
+        LOGS(_log, LOG_LVL_ERROR, context_ << "rows.size(): " << resultSet.rows.size()
+                << " fields.size(): " << resultSet.fields.size()
+                << " fields[0].name: " << (resultSet.fields.size() > 0 ? resultSet.fields[0].name : "")
+                << " fields[1].name: " << (resultSet.fields.size() > 1 ? resultSet.fields[1].name : ""));
+        return !succeeded;
+    }
+    if (resultSet.rows.empty()) {
+        // IMPORTANT: Each tables is required to have a representation in the statistics
+        // even if it has 0 rows. The default transaction 0 will be used in this case.
+        counters[internalTable][0].push_back(0);
+    } else {
+        for (auto&& row: resultSet.rows) {
+            if (row.nulls[0] || row.nulls[1]) {
+                reportResultThatHas("unexpected NULL values");
+                return !succeeded;
+            }
+            try {
+                TransactionId const transactionId = lsst::qserv::replica::stoui(row.cells[0]);
+                size_t const numRows = std::stoul(row.cells[1]);
+                counters[internalTable][transactionId].push_back(numRows);
+            } catch (...) {
```

Comment:
This is a very generic catch, but the code seems to assume it's due to badly formatted numbers?

Reply:
Yes, because the only method where it's expected to fail is calling:
```
lsst::qserv::replica::stoui
std::stoul
```
Dig into fine details of a problem makes very little sense in this context. The catch was just meant to ensure the result set received from a worker is not corrupt.

Reply:
This is not quite true even for current code, `operator[]` and `push_back` may throw if memory allocation fails.

Reply:
👍